File: android/Reguerta/app/src/main/java/com/reguerta/user/domain/notifications/NotificationRepository.kt
package com.reguerta.user.domain.notifications

import com.reguerta.user.domain.access.Member

interface NotificationRepository {
    suspend fun getNotificationsFor(member: Member): List<NotificationEvent>

    suspend fun getReadNotificationIds(memberId: String): Set<String>

    suspend fun markNotificationsRead(
        memberId: String,
        notificationIds: Set<String>,
        readAtMillis: Long,
    )

    suspend fun sendNotification(event: NotificationEvent): NotificationEvent
}

File: ios/Reguerta/Reguerta/Domain/Notifications/NotificationRepository.swift
import Foundation

protocol NotificationRepository: Sendable {
    func notifications(visibleTo member: Member, environment: SessionEnvironment) async throws -> [NotificationEvent]
    func allNotifications(environment: SessionEnvironment) async throws -> [NotificationEvent]
    func readNotificationIds(memberId: String, environment: SessionEnvironment) async throws -> Set<String>
    func markNotificationsRead(
        memberId: String,
        notificationIds: [String],
        readAtMillis: Int64,
        environment: SessionEnvironment
    ) async throws
    func send(event: NotificationEvent, environment: SessionEnvironment) async throws -> NotificationEvent
}
